internal/server/pool_collector.go

```diff
@@ -0,0 +1,100 @@
+package server
+
+import "github.com/prometheus/client_golang/prometheus"
+
+type poolCollector struct {
+	p *Pool
+
+	maxOpenConnections *prometheus.Desc
+
+	openConnections  *prometheus.Desc
+	inUseConnections *prometheus.Desc
+	idleConnections  *prometheus.Desc
+
+	waitCount         *prometheus.Desc
+	waitDuration      *prometheus.Desc
```

Comment:
Can we add the unit to the name itself?
```suggestion
	waitDurationSec      *prometheus.Desc
```